index.js

```diff
@@ -35,16 +32,19 @@ Quickbooks.BASE_URL                   = 'https://sandbox.api.intuit.com/quickboo
  * @param debug - boolean flag to turn on logging of HTTP requests, including headers and body
  * @constructor
  */
-function Quickbooks(consumerKey, consumerSecret, token, tokenSecret, realmId, useSandbox, debug) {
-  var prefix           = _.isObject(consumerKey) ? 'consumerKey.' : ''
-  this.consumerKey     = eval(prefix + 'consumerKey')
-  this.consumerSecret  = eval(prefix + 'consumerSecret')
-  this.token           = eval(prefix + 'token')
-  this.tokenSecret     = eval(prefix + 'tokenSecret')
-  this.realmId         = eval(prefix + 'realmId')
-  this.useSandbox      = eval(prefix + 'useSandbox')
-  this.debug           = eval(prefix + 'debug')
-  this.endpoint        = this.useSandbox ? Quickbooks.BASE_URL : Quickbooks.BASE_URL.replace('sandbox.', '')
+function Quickbooks({consumerKey, consumerSecret, token, tokenSecret, realmId, refreshToken, oauthversion, useSandbox, debug}) {
```

Comment:
Will converting the constructor arguments to a single map break existing clients?

Reply:
I think yes. But previous constructor signature failed when I ran your tests since you used Object spreading style to instantiate. So I changed. 

 `qbo = new QuickBooks(config);`
Maybe you can revert it to old style.

Reply:
see here for example https://github.com/mcohen01/quickbooks/blob/567f4142158969647bc1d47ad72bddb7e3229f0a/test/index.js#L7

Reply:
we should fix the tests then before we break existing clients

Reply:
yup fixed in this [pull request](https://github.com/mcohen01/quickbooks/pull/6/commits/1dbe035a5bc2e6a8a44877eb2c087a05929bcd60)